featuretools/entityset/entityset.py

```diff
@@ -1168,124 +1164,127 @@ def add_interesting_values(self, max_values=5, verbose=False, dataframe_name=Non
-    #     dataframe.
-    #     """
-    #     dt = self[entity_id]
-    #     if is_instance(df, ks, 'DataFrame') and isinstance(time_last, np.datetime64):
-    #         time_last = pd.to_datetime(time_last)
-    #     if dt.time_index:
-    #         df_empty = df.empty if isinstance(df, pd.DataFrame) else False
-    #         if time_last is not None and not df_empty:
-    #             if include_cutoff_time:
-    #                 df = df[df[dt.time_index] <= time_last]
-    #             else:
-    #                 df = df[df[dt.time_index] < time_last]
-    #             if training_window is not None:
-    #                 training_window = _check_timedelta(training_window)
-    #                 if include_cutoff_time:
-    #                     mask = df[dt.time_index] > time_last - training_window
-    #                 else:
-    #                     mask = df[dt.time_index] >= time_last - training_window
-    #                 if dt.last_time_index is not None:
-    #                     lti_slice = dt.last_time_index.reindex(df.index)
-    #                     if include_cutoff_time:
-    #                         lti_mask = lti_slice > time_last - training_window
-    #                     else:
-    #                         lti_mask = lti_slice >= time_last - training_window
-    #                     mask = mask | lti_mask
-    #                 else:
-    #                     warnings.warn(
-    #                         "Using training_window but last_time_index is "
-    #                         "not set on entity %s" % (dt.id)
-    #                     )
-
-    #                 df = df[mask]
-
-    #     for secondary_time_index, columns in dt.secondary_time_index.items():
-    #         # should we use ignore time last here?
-    #         df_empty = df.empty if isinstance(df, pd.DataFrame) else False
-    #         if time_last is not None and not df_empty:
-    #             mask = df[secondary_time_index] >= time_last
-    #             if isinstance(df, dd.DataFrame):
-    #                 for col in columns:
-    #                     df[col] = df[col].mask(mask, np.nan)
-    #             elif is_instance(df, ks, 'DataFrame'):
-    #                 df.loc[mask, columns] = None
-    #             else:
-    #                 df.loc[mask, columns] = np.nan
-
-    #     return df
-
-    # def query_by_values(self, entity_id, instance_vals, variable_id=None, columns=None,
-    #                     time_last=None, training_window=None, include_cutoff_time=True):
-    #     """Query instances that have variable with given value
+    def _handle_time(self, dataframe_id, df, time_last=None, training_window=None, include_cutoff_time=True):
+        """
+        Filter a dataframe for all instances before time_last.
+        If the dataframe does not have a time index, return the original
+        dataframe.
+        """
+        dt = self[dataframe_id]
```

Comment:
just wondering - why have we been using `dt` here? It doesn't seem like it's a reference to DataTables or anything, and I think it's getting used interchangeably with `df` here, so I'm not sure I understand its purpose

Reply:
Yeah, that is unrelated to the Woodwork integration and that variable name is carried over from the code in `main`. The only reason that is used as far as I can tell is to access the various "entity" attributes like `index` and `time_index` which are needed for the filtering operations since those values aren't necessarily present on the dataframe that gets passed into the method.

Reply:
Okay, I see the purpose; it's kind of like the way that we use typing information in some of the Woodwork statistics utils.

If we're only using it to access woodwork typing information, I think my suggestion would be to just pass the schema into the method. We don't use the `dataframe_name` for anything other than getting the `dt` variable, and passing around the schema is what we do in `_make_categorical_for_mutual_info` and `_replace_nans_for_mutual_info`, which are decent analogs for `_handle_time`.

Reply:
Yep, we could do that. My only hesitation is related to whether that change obscures a little by what the function call is doing. Now by passing the entity name it is clear to me that we are handling time for the specified entity/dataframe, and passing a schema instead seems to make that a little less obvious to me, although I don't think that's a huge problem since this is a private method anyway.

One other option might be to continue to pass in the dataframe name but grab the schema and use that in place of `dt`. So instead of 
```
dt = self[dataframe_name]
```

we could use
```
schema = self[dataframe_name].ww.schema
```
and then update all the following references to `dt`, which I think would make it more clear what is happening inside this function.

I don't have a strong opinion here, but I think we can make some changes to clean this up a bit.

Reply:
Using the dataframe name to get the schema seems like a good combination!